src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/Docker.java

```diff
@@ -212,7 +244,7 @@ public String runDetached(String image, String workdir, Map<String, String> volu
         }
 
         for (Map.Entry<String, String> e : environment.entrySet()) {
-            if ("HOSTNAME".equals(e.getKey())) {
+            if ("HOSTNAME".equals(e.getKey()) || protectedEnvironmentVariables.contains(e.getKey())) {
```

Comment:
Are you recommending that I add a check here to ensure protectedEnvironmentVariables is not null?